Sources_v3/Workers/Background/NewChannelQueryUpdater.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+
+/// After creating new channel it's not observed cause it's not linked to any ChannelListQuery.
+/// The only job of `NewChannelQueryUpdater` is to find whether new channel belongs to any of the exsisting queries
+/// and link it to the channel if so.
+///     1. This worker observers DB for the insertations of the new `ChannelDTO`s without any linked queries.
+///     2. When new channel is found, all exsisting queries are fetched from DB and we modify exsiting queries filters so
+///     in response for `update(channelListQuery` request new channel will be returned if it is part of the original query filter.
+///     3. After sending `update(channelListQuery` for all queries `ChannelListQueryUpdater` does the job of linking
+///     corresponding queries to the channel.
+final class NewChannelQueryUpdater<ExtraData: ExtraDataTypes>: Worker, NSFetchedResultsControllerDelegate {
+    typealias DatabaseObserver = ListDatabaseObserver<ChannelDTO, ChannelDTO>
+    
+    private let environment: Environment
+        
+    private lazy var channelListQueryUpdater: ChannelListQueryUpdater<ExtraData> = self.environment
+        .createChannelListQueryUpdater(database,
+                                       webSocketClient,
+                                       apiClient)
+    
+    private lazy var channelsObserver: ListDatabaseObserver = .init(context: self.database.writableContext,
+                                                                    fetchRequest: ChannelDTO.newChannelsFetchRequest)
+    
+    private var queries: [ChannelListQueryDTO] {
+        do {
+            let queries = try database.backgroundReadOnlyContext
+                .fetch(NSFetchRequest<ChannelListQueryDTO>(entityName: ChannelListQueryDTO.entityName))
+            return queries
+        } catch {
+            log.error("Internal error: Failed to fetch [ChannelListQueryDTO].")
+        }
+        return []
+    }
+    
+    init(database: DatabaseContainer, webSocketClient: WebSocketClient, apiClient: APIClient, env: Environment) {
+        environment = env
+        super.init(database: database, webSocketClient: webSocketClient, apiClient: apiClient)
+        
+        startObserving()
+    }
+    
+    override convenience init(database: DatabaseContainer, webSocketClient: WebSocketClient, apiClient: APIClient) {
+        self.init(database: database, webSocketClient: webSocketClient, apiClient: apiClient, env: .init())
+    }
```

Comment:
Ugh, I see there's a problem with initializers from the superclass. 🤔 Any idea how to solve this?